packages/plugin-data/src/api/data/tree/importApi.ts

```diff
@@ -0,0 +1,496 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+
+import * as path from 'path';
+import * as util from 'util';
+
+import {
+  Dictionary,
+  isString,
+  isObject,
+  isArray,
+  getNumber,
+  getString,
+  AnyJson,
+  JsonMap,
+  JsonCollection,
+} from '@salesforce/ts-types';
+import { fs, Logger, Org, SfdxError, SchemaValidator } from '@salesforce/core';
+import { sequentialExecute } from './executors';
+
+const importPlanSchemaFile = path.join(__dirname, '..', '..', '..', '..', 'schema', 'dataImportPlanSchema.json');
+
+const sobjectTreeApiPartPattern = '%s/services/data/v%s/composite/tree/%s';
+const jsonContentType = 'application/json';
+const xmlContentType = 'application/xml';
+const jsonRefRegex = /[.]*["|'][A-Z0-9_]*["|'][ ]*:[ ]*["|']@([A-Z0-9_]*)["|'][.]*/gim;
+const xmlRefRegex = /[.]*<[A-Z0-9_]*>@([A-Z0-9_]*)<\/[A-Z0-9_]*[ID]>[.]*/gim;
+
+const INVALID_DATA_IMPORT_ERR_NAME = 'InvalidDataImport';
+
+interface DataImportComponents {
+  instanceUrl: string;
+  saveRefs?: boolean;
+  resolveRefs?: boolean;
+  refMap: Map<string, string>;
+  filepath: string;
+  contentType?: string;
+}
+
+export interface ImportConfig {
+  sobjectTreeFiles?: string[];
+  contentType?: string;
+  plan?: string;
+}
+
+interface ResponseRefs {
+  referenceId: string;
+  id: string;
+}
+
+export interface ImportResults {
+  responseRefs?: ResponseRefs[];
+  sobjectTypes?: AnyJson;
+  errors?: string[];
+}
+
+/**
+ * Imports data into an org that was exported to files using the export API.
+ */
+export class ImportApi {
+  private logger: Logger;
+  private responseRefs: ResponseRefs[] = [];
+  private sobjectUrlMap: Map<string, string>;
+  private schemaValidator: SchemaValidator;
+  private sobjectTypes: any = {}; // hash of sobject { referenceId: type }
+
+  private config!: ImportConfig;
+
+  // A JsonArray but we need to provide a better type to work with it properly.
+  private importPlanConfig: any;
+
+  public constructor(private readonly org: Org) {
+    this.logger = Logger.childFromRoot(this.constructor.name);
+
+    this.sobjectUrlMap = new Map<string, string>();
+    this.schemaValidator = new SchemaValidator(this.logger, importPlanSchemaFile);
+  }
+
+  /**
+   * Inserts given SObject Tree content into given target Org.
+   *
+   * Validation and options fix-up done in execute() instead
```

Comment:
is this referencing the old `execute` methods that were on commands if so it should be `run` now right?

Reply:
Oh good catch.  I renamed some methods within this class.  There used to be one called `execute`.  This comment should just go away because it refers to very legacy things.